controllers/datafoundationclient.go

```diff
@@ -47,16 +48,27 @@ func dfcSetupWatches(controllerBuilder *builder.Builder) {
 		),
 	)
 	controllerBuilder.
-		Owns(&opv1a1.ClusterServiceVersion{}, csvPredicates).
-		Owns(&ocsclient.StorageClient{}).
-		Owns(&ocsclient.StorageClassClaim{})
+		Owns(&opv1a1.ClusterServiceVersion{}, csvPredicates)
+
+	if utils.DoesCRDsExists("storageclients.ocs.openshift.io") {
+		doesCRDExistsAtStart = true
+		controllerBuilder.
+			Owns(&ocsclient.StorageClient{}).
+			Owns(&ocsclient.StorageClassClaim{})
+	}
 }
 
 func DFCAddToScheme(scheme *runtime.Scheme) {
 	utilruntime.Must(ocsclient.AddToScheme(scheme))
 }
 
 func dfcReconcile(offeringReconciler *ManagedFusionOfferingReconciler, offering *v1alpha1.ManagedFusionOffering) (ctrl.Result, error) {
+	if !doesCRDExistsAtStart {
+		if utils.DoesCRDsExists("storageclients.ocs.openshift.io") {
+			return ctrl.Result{}, errRestart
```

Comment:
I don't think this is an error. We are restarting because the offering installation succeeded.

Reply:
done